minetest/minetest
# Add bumpmapping and parallax occlusion testing nodes to the devtest game

Goal: Simple nodes for testing bumpmapping and parallax occlusion code changes

How does it work:
The alpha opacity in the normalmap texture is the heightmap for Parallax Occlusion.
The sRGB values in the normalmap store the normal vector, e.g. #8080ff is (2*(128/256-0.5), 2*(128/256-0.5), 2*(255/256-0.5)) = (0,0,1) (the default surface normal).

Does it resolve any reported issue: It could indirectly help to solve related issues.

If not a bug fix, why is this PR needed? What usecases does it solve: The height pyramid normals have 45° rotation, so they can be used to validate if bumpmapping, or a changed bumpmapping implementation, works correctly.

I do not understand what bumpmapping in minetest does:
![2](https://user-images.githubusercontent.com/3192173/71368662-ca7acb80-25a8-11ea-9468-a9969a4b43be.png)
Relief mapping (one of the Parallax Occlusion methods):
![1](https://user-images.githubusercontent.com/3192173/71368653-c5b61780-25a8-11ea-863f-fbe82791d978.png)


## To do

This PR is Ready for Review.


## How to test

Enable Bumpmapping and/or Parallax Occlusion, place the nodes and see what happens.
